pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -282,6 +282,15 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("A volume match does not exist for persistent claim: %s", claim.Name)
 		}
 
+		// create a reference to the claim and assign it to the volume being bound.
+		// the volume is a pointer and assigning the reference fixes a race condition where another
+		// claim might match this volume but before the claimRef is persistent in the next case statement
+		claimRef, err := api.GetReference(claim)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
+		}
+		volume.Spec.ClaimRef = claimRef
```

Comment:
Agreed.  For this to work correctly, I believe I need to change the official bind, as noted above.

Coincidentally, I have a TODO in https://github.com/kubernetes/kubernetes/pull/14537 to ask @thockin about changing the bind.  

This fix works narrowly, where OpenShift templates can make multiple claims at once.  This fix shrinks the race condition but doesn't eliminate it.